tests/src/Functional/PrepaidBalanceReportsDownloadFunctionalTest.php

```diff
@@ -0,0 +1,155 @@
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    $this->account = $this->createAccount([
+      'view mint prepaid reports',
+      'download prepaid balance reports',
+    ]);
+
+    $this->queueOrg();
+
+    $this->drupalLogin($this->account);
+  }
+
+  /**
+   * Test for an account with no reports.
+   *
+   * @throws \Behat\Mink\Exception\ResponseTextException
+   */
+  public function testAccountWithNoReports() {
+    $this->queueMockResponses([
+      'get-prepaid-balances',
+      'get-supported-currencies',
+      'get-billing-documents-months',
+    ]);
+
+    $this->drupalGet(Url::fromRoute('apigee_monetization.billing', [
+      'user' => $this->account->id(),
+    ]));
+
+    $this->queueMockResponses([
+      'get-prepaid-balances',
+      'get-supported-currencies',
+      'get-billing-documents-months',
+      'post-prepaid-balance-reports-empty',
+    ]);
+
+    $edit = ['currency' => 'test', 'year' => '2018', 'month_2018' => '2018-9'];
+    $this->submitForm($edit, 'Download CSV');
+
+    $this->assertSession()
+      ->pageTextContains('There are no prepaid balance reports for account TEST (Test) for September 2018.');
+  }
+
+  /**
+   * Test for a date with no reports.
+   *
+   * @throws \Behat\Mink\Exception\ResponseTextException
+   */
+  public function testDateWithNoReport() {
+    $this->queueMockResponses([
+      'get-prepaid-balances',
+      'get-supported-currencies',
+      'get-billing-documents-months',
+    ]);
+
+    $this->drupalGet(Url::fromRoute('apigee_monetization.billing', [
+      'user' => $this->account->id(),
+    ]));
+
+    $this->queueMockResponses([
+      'get-prepaid-balances',
+      'get-supported-currencies',
+      'get-billing-documents-months',
+      'post-prepaid-balance-reports-empty',
+    ]);
+
+    $edit = ['currency' => 'usd', 'year' => '2018', 'month_2018' => '2018-8'];
+    $this->submitForm($edit, 'Download CSV');
+
+    $this->assertSession()
+      ->pageTextContains('There are no prepaid balance reports for account USD (United States Dollars) for August 2018.');
+  }
+
+  /**
+   * Test the prepaid balance download form for a valid account and date.
+   *
+   * @throws \Behat\Mink\Exception\ExpectationException
+   */
+  public function testValidAccountAndDate() {
+    $this->queueMockResponses([
+      'get-prepaid-balances',
+      'get-supported-currencies',
+      'get-billing-documents-months',
+    ]);
+
+    $this->drupalGet(Url::fromRoute('apigee_monetization.billing', [
+      'user' => $this->account->id(),
+    ]));
+
+    $this->queueMockResponses([
+      'get-prepaid-balances',
+      'get-supported-currencies',
+      'get-billing-documents-months',
+      'post-prepaid-balance-reports',
+    ]);
+
+    $edit = ['currency' => 'usd', 'year' => '2018', 'month_2018' => '2018-9'];
+    $this->submitForm($edit, 'Download CSV');
+
+    $filename = "prepaid-balance-report-2018-9.csv";
+    $this->assertSession()->statusCodeEquals(Response::HTTP_OK);
```

Comment:
Unfortunately not all webdrivers support `getStatusCode()`. Using Selenium2Driver instead of Goutte for functional tests causes this to fail.